CRM/Contribute/Form/Contribution.php

```diff
@@ -2071,6 +2059,17 @@ protected function submit($submittedValues, $action, $pledgePaymentID) {
           }
         }
         $lineItem[$this->_priceSetId][$itemId] = $item;
+        if (count($lineItems) === 1 && $item['entity_table'] === 'civicrm_participant') {
+          // CRM-10117 update the line items for participants.
+          // We can only do this for exactly 1 as otherwise we do not know how to allocate.
+          // @todo - we should handle this in LineItem::create() when we update the line item
+          // amount. The only issue is we need to double check how multiple registrations
+          // against 1 line item work.
+          Participant::update(FALSE)
```

Comment:
I presume the answer is no but is there any way this could be called with the `civi_event` extension / component not enabled ?

Reply:
@seamuslee001 oh wow - that hurts the brain - because they would need to have line items with participant records and then disable it - we should block disabling in that case!! I think it's too edge

Reply:
@seamuslee001 I could wrap in `CRM_Core_Component::isEnabled('CiviEvent')` - surely that would imply the extension is enabled?

Reply:
It would but it feels like quite the edge case, i'm ok with this otherwise

Reply:
thanks @seamuslee001